thrust/thrust/zip_function.h

```diff
@@ -95,54 +95,40 @@ _CCCL_HOST_DEVICE auto apply_impl(Function&& func, Tuple&& args, index_sequence<
  *  #include <thrust/zip_function.h>
  *
  *  struct SumTuple {
- *    float operator()(Tuple tup) {
- *      return std::get<0>(tup) + std::get<1>(tup) + std::get<2>(tup);
+ *    float operator()(auto tup) const {
+ *      return thrust::get<0>(tup) + thrust::get<1>(tup) + thrust::get<2>(tup);
```

Comment:
For `std::get` to compile we would need to implement the tuple protocol for `tuple_of_iterator_references`. I was too lazy to do that including tests.